global_helpers/panther_oss_helpers.py

```diff
@@ -163,265 +162,88 @@ def resource_lookup(resource_id: str) -> Dict[str, Any]:
     return response["Item"]["attributes"]
 
 
-# Helper functions for accessing Dynamo key-value store.
-#
-# Keys can be any string specified by rules and policies,
-# values are integer counters and/or string sets.
-#
-# Use kv_table() if you want to interact with the table directly.
-_KV_TABLE = None
-_COUNT_COL = "intCount"
-_STRING_SET_COL = "stringSet"
-_DICT_COL = "dictionary"
-_TTL_COL = "expiresAt"
-
-
-def kv_table() -> boto3.resource:
-    """Lazily build key-value table resource"""
-    # pylint: disable=global-statement
-    global _KV_TABLE
-    if not _KV_TABLE:
-        # pylint: disable=no-member
-        _KV_TABLE = boto3.resource(
-            "dynamodb",
-            endpoint_url="https://dynamodb" + FIPS_SUFFIX if FIPS_ENABLED else None,
-        ).Table("panther-kv-store")
-    return _KV_TABLE
-
-
 def ttl_expired(response: dict) -> bool:
-    """Checks whether a response from the panther-kv table has passed it's TTL date"""
-    # This can be used when the TTL timing is very exacting and DDB's cleanup is too slow
-    expiration = response.get("Item", {}).get(_TTL_COL, 0)
-    return expiration and float(expiration) <= (datetime.now()).timestamp()
+    """Global `ttl_expired` is DEPRECATED.
+    Instead, use `panther_detection_helpers.caching.ttl_expired`."""
+    return caching.ttl_expired(response)
 
 
 def get_counter(key: str, force_ttl_check: bool = False) -> int:
-    """Get a counter's current value (defaulting to 0 if key does not exist)."""
-    response = kv_table().get_item(
-        Key={"key": key},
-        ProjectionExpression=f"{_COUNT_COL}, {_TTL_COL}",
-    )
-    if force_ttl_check and ttl_expired(response):
-        return 0
-    return response.get("Item", {}).get(_COUNT_COL, 0)
+    """Global `get_counter` is DEPRECATED.
+    Instead, use `panther_detection_helpers.caching.get_counter`."""
+    return caching.get_counter(key=key, force_ttl_check=force_ttl_check)
 
 
 def increment_counter(key: str, val: int = 1) -> int:
-    """Increment a counter in the table.
-
-    Args:
-        key: The name of the counter (need not exist yet)
-        val: How much to add to the counter
-
-    Returns:
-        The new value of the count
-    """
-    response = kv_table().update_item(
-        Key={"key": key},
-        ReturnValues="UPDATED_NEW",
-        UpdateExpression="ADD #col :incr",
-        ExpressionAttributeNames={"#col": _COUNT_COL},
-        ExpressionAttributeValues={":incr": val},
-    )
-
-    # Numeric values are returned as decimal.Decimal
-    return response["Attributes"][_COUNT_COL].to_integral_value()
+    """Global `increment_counter` is DEPRECATED.
+    Instead, use `panther_detection_helpers.caching.increment_counter`."""
+    return caching.increment_counter(key=key, val=val)
```

Comment:
Did we want to add the optional ttl arg here as well?

Reply:
I thought about doing that. I think no because we are trying to leave these as is just so we don't break anything, but we really want to encourage users to use the caching module directly

Reply:
Worst case we can always add it later if someone really needs it